starknet_py/devnet_utils/devnet_client.py

```diff
@@ -0,0 +1,309 @@
+        """
+
+        super().__init__(node_url=node_url, session=session)
+        self._devnet_client = RpcHttpClient(
+            url=node_url, session=session, method_prefix="devnet"
+        )
+
+    async def impersonate_account(self, address: Hash):
+        """
+        Impersonate the given account.
+        For impersonation to work, Devnet needs to be run in forking mode.
+
+        :param address: Address of the account contract.
+        """
+
+        await self._devnet_client.call(
+            method_name="impersonateAccount",
+            params={"account_address": _to_rpc_felt(address)},
+        )
+
+    async def stop_impersonate_account(self, address: Hash):
+        """
+        Stop impersonating the given account.
+
+        :param address: Address of the account contract.
+        """
+
+        await self._devnet_client.call(
+            method_name="stopImpersonateAccount",
+            params={"account_address": _to_rpc_felt(address)},
+        )
+
+    async def auto_impersonate(self):
+        """
+        Enables automatic account impersonation.
+        Every account that does not exist in the local state will be impersonated.
+        For impersonation to work, Devnet needs to be run in forking mode.
+        """
+
+        await self._devnet_client.call(method_name="autoImpersonate")
+
+    async def stop_auto_impersonate(self):
+        await self._devnet_client.call(method_name="stopAutoImpersonate")
+
+    async def mint(
+        self, address: Hash, amount: int, unit: Union[PriceUnit, str] = PriceUnit.WEI
+    ) -> MintResponse:
+        """
+        Mint tokens to the given address.
+
+        :param address: Address of the account contract.
+        :param amount: Amount of tokens to mint. Must be integer.
+        :param unit: Literals `"FRI"` or `"WEI"`, default to `"WEI"`.
+        """
+
+        res = await self._devnet_client.call(
+            method_name="mint",
+            params={
+                "address": _to_rpc_felt(address),
+                "amount": amount,
+                "unit": unit.upper() if isinstance(unit, str) else unit.value,
+            },
+        )
+
+        return cast(MintResponse, MintSchema().load(res))
+
+    async def get_account_balance(
+        self,
+        address: Hash,
+        unit: Union[PriceUnit, str] = PriceUnit.WEI,
+        block_tag: Optional[str] = None,
+    ) -> BalanceRecord:
+        """
+        Get the balance of the given account.
+
+        :param address: Address of the account contract.
+        :param unit: Literals `"FRI"` or `"WEI"` defaults to `"WEI"`.
+        :param block_tag: Literals `"pending"` or `"latest"`, defaults to `"latest"`.
+        """
+
+        res = await self._devnet_client.call(
+            method_name="getAccountBalance",
+            params={
+                "address": _to_rpc_felt(address),
+                "unit": unit.upper() if isinstance(unit, str) else unit.value,
+                "block_tag": block_tag,
```

Comment:
Does it work if `block_tag` is `None`?

Reply:
Devnet by default takes the "latest" block in this scenario as it is written in the docstring, but I will set the default value to "latest" to improve readability.